dask/dataframe/io/parquet/arrow.py

```diff
@@ -107,523 +114,106 @@ def _index_in_schema(index, schema):
-                            file_row_group_column_stats = {}
-                            break
-
-                    if single_rg_parts:
-                        to_ts = column.statistics.logical_type.type == "TIMESTAMP"
-                        s["columns"].append(
-                            {
-                                "name": name,
-                                "min": cmin if not to_ts else pd.Timestamp(cmin),
-                                "max": cmax if not to_ts else pd.Timestamp(cmax),
-                                "null_count": cnull,
-                            }
-                        )
-                    else:
-                        cstats += [cmin, cmax, cnull]
-                    cmax_last[name] = cmax
-                else:
-
-                    if no_filters and column.num_values > 0:
-                        # We are collecting statistics for divisions
-                        # only (no filters) - Lets bail.
-                        gather_statistics = False
-                        file_row_group_stats = {}
-                        file_row_group_column_stats = {}
-                        break
+    # Get/transate filters
+    ds_filters = None
+    if filters is not None:
+        ds_filters = pq._filters_to_expression(filters)
+
+    # Get all partition keys (without filters) to populate partition_obj
+    partition_obj = []  # See `partition_info` description below
+    categories, pkeys = _get_all_partition_keys(ds)
+    partition_names = list(categories)
+    for name in partition_names:
+        partition_obj.append(PartitionObj(name, categories[name]))
+
+    # Split by row-groups and apply filters
+    partition_keys = {}  # See `partition_info` description below
+    metadata = []  # List of row-group fragments
+    # Loop over file fragments
+    for file_frag in ds.get_fragments(ds_filters):
+        # If valid_paths is not None, the user passed in a list
+        # of files containing a _metadata file.  Since we used
+        # the _metadata file to generate our dataset object , we need
+        # to ignore any file fragments that are not in the list.
+        if valid_paths and file_frag.path.split(fs.sep)[-1] not in valid_paths:
+            continue
+        # Store (filtered) partition keys
+        if pkeys:
+            partition_keys[file_frag.path] = [
+                (name, pkeys[file_frag.path].get(name, None))
+                for name in partition_names
+            ]
 
-                    if single_rg_parts:
-                        s["columns"].append({"name": name})
-                    else:
-                        cstats += [None, None, None]
+        if split_row_groups is False and ds_filters is None:
+            # Avoid row-group splitting.
+            # NOTE: We may NOT want to do this if we are filtering.
+            # The resulting divisions (if there is an index) will be
+            # calculated with UNFILTERED row-groups.  This is probably
```

Comment:
I have been thinking about this particular code path for a bit today.  I (hopefully) clarified the comment a bit, and revised the code to also avoid splitting when divisions will not be needed later (even if there are filters) - With that said, I now realize that much of this logic may be overkill... 

There is no doubt that it can be a problem to calculate divisions with "unfiltered" statistics.  For example, it may be the case that the "to-be" index column is not monotonically increasing until after filters are applied.  In this case, the `read_parquet` call will ultimately result in unknown divisions if we do not split the fragments by row-group (see new test: `test_pyarrow_filter_divisions`).

Although we are now handling this case, I'm not sure it makes sense to prioritize accurate divisions over performance.  If I undserstand correctly, we are not actually introducing any regressions in functionality compared to the legacy engine.  Prior to this PR, the resulting divisions would have been exactly the same, the only difference is that there was previously more (unfiltered) data in the final output partitions.  Even if the user were to perform follow-up row-wise filtering after the read, the "filtered" divisions wouldn't be correct:

```python
In [1]: import pandas as pd
   ...: import dask.dataframe as dd
   ...: df = pd.DataFrame({"a": range(10)})
   ...: ddf = dd.from_pandas(df, npartitions=2)
   ...: ddf.divisions
Out[1]: (0, 5, 9)

In [2]: ddf = ddf[ddf["a"] < 9]
   ...: ddf.divisions
Out[2]: (0, 5, 9)
```

In summary, I am thinking that one opportunity to simplify `ArrowDatasetEngine` is to allow divisions to reflect the unfiltered data.  This would simplify the code, and may improve performance anyway...  Still thinking about this.

Reply:
Yeah, it's indeed a bit of a trade-off, and I don't have enough real-world experience with divisions to know how important it is that they reflect the filtered data, or whether it is fine to reflect non-filtered data. 

I think I mentioned it before, but a possible solution could be to split row groups (and filter row groups) in order to collect statistics, but not split it from `read_partition` (either by also keeping track of the unsplitted fragment, or by recreating the fragment with the filtered row group ids). 
Now, this might complicate the code further, rather than simplify it though ..  But that feels "most" correct

Reply:
> I think I mentioned it before, but a possible solution could be to split row groups (and filter row groups) in order to collect statistics, but not split it from read_partition (either by also keeping track of the unsplitted fragment, or by recreating the fragment with the filtered row group ids).
Now, this might complicate the code further, rather than simplify it though .. But that feels "most" correct

You did recommend something like this earlier and I missed it - This is a great suggestion, and is now what I am trying to do in the latest code 👍